File: WPLLib/WPLRunTest/MainForm.vb
Option Strict On

Imports System.Xml
Imports System.Text
Imports WPLLib

Public Class MainForm

    Dim media As ArrayList
    Dim wplclass As New WPLControl
    Private Sub SaveBT_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles SaveBT.Click
        Dim sfd As New SaveFileDialog()

        sfd.FileName = ""
        sfd.InitialDirectory = "C:\"
        sfd.Filter = _
         "Windows Play List(*.wpl)|*.wpl|すべてのファイル(*.*)|*.*"
        sfd.FilterIndex = 1
        sfd.Title = "保存先のファイルを選択してください"
        sfd.RestoreDirectory = True
        sfd.OverwritePrompt = True
        sfd.CheckPathExists = True

        If sfd.ShowDialog() = DialogResult.OK Then
            wplclass.WPLCreate(sfd.FileName, media)
        End If
    End Sub

    Private Sub LoadBT_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles LoadBT.Click
        Dim ofd As New OpenFileDialog()

        ofd.FileName = ""
        ofd.InitialDirectory = "C:\"
        ofd.Filter = _
            "Windows Play List(*.wpl)|*.wpl|すべてのファイル(*.*)|*.*"
        ofd.FilterIndex = 1
        ofd.Title = "開くファイルを選択してください"
        ofd.RestoreDirectory = True
        ofd.CheckFileExists = True
        ofd.CheckPathExists = True

        If ofd.ShowDialog() = DialogResult.OK Then
            media = wplclass.WPLLoad(ofd.FileName)
            For i = 0 To media.Count - 1
                MusicList.Items.Add(media.Item(i).ToString())
            Next
        End If
    End Sub
End Class

File: WPLLib/WPLRunTest-CS/MainForm.cs
using System;
using System.Windows.Forms;
using WPLLib;


namespace WPLRunTest_CS
{
    public partial class MainForm : Form
    {
        public MainForm()
        {
            InitializeComponent();
        }
        
        private void SaveBT_Click(object sender, EventArgs e)
        {
            SaveFileDialog sfd = new SaveFileDialog();

            sfd.FileName = "";
            sfd.InitialDirectory = @"C:\";
            sfd.Filter = "Windows Play List(*.wpl)|*.wpl|All Files(*.*)|*.*";
            sfd.FilterIndex = 1;
            sfd.Title = "Please select a file destination";
            sfd.RestoreDirectory = true;
            sfd.OverwritePrompt = true;
            sfd.CheckPathExists = true;

            //Create WPL File.
            //First argument: Path of file to be saved
            //Second argument: ArrayList of the MediaList
            if (sfd.ShowDialog() == DialogResult.OK)
            {
                WPLLib.Writer wplWrite = new Writer(sfd.FileName);
                foreach (string filePath in MusicList.Items)
                {
                    wplWrite.Add(filePath);
                }
                wplWrite.Write();
            }
        }
        private void LoadBT_Click(object sender, EventArgs e)
        {
            OpenFileDialog ofd = new OpenFileDialog();

            ofd.FileName = "";
            ofd.InitialDirectory = @"C:\";
            ofd.Filter = "Windows Play List(*.wpl)|*.wpl|All Files(*.*)|*.*";
            ofd.FilterIndex = 1;
            ofd.Title = "Please select a file to open";
            ofd.RestoreDirectory = true;
            ofd.CheckFileExists = true;
            ofd.CheckPathExists = true;

            if (ofd.ShowDialog() == DialogResult.OK)
            {
                WPLLib.Reader wplReader = new Reader(ofd.FileName);
                wplReader.Read();
                WPLLib.MediaArray medias = wplReader.GetMediaArray();

                foreach (Media media in medias)
                {
                    MusicList.Items.Add(media.FilePath);
                }
            }
        }

        private void MusicList_DragEnter(object sender, DragEventArgs e)
        {
            if (e.Data.GetDataPresent(DataFormats.FileDrop))
            {
                e.Effect = DragDropEffects.Copy;
            }
            else
            {
                e.Effect = DragDropEffects.None;
            }
        }
        private void MusicList_DragDrop(object sender, DragEventArgs e)
        {
            string[] filePaths = (string[])e.Data.GetData(DataFormats.FileDrop, false);
            foreach (string filePath in filePaths)
            {
                MusicList.Items.Add(filePath);
            }
        }
    }
}
